api_types.ts

```diff
@@ -494,6 +494,15 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    * so an end user must have both sets of permissions.
    */
   readonly extraOAuthScopes?: string[];
+
+  /**
+   * TODO(patrick): Without this set, should we allow all or only DEFAULT?
+   *
+   * Ignored if connectionRequirement is None.
```

Comment:
When we add validation should probably validate that you can't set these together.